x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/CcrRepositoryIT.java

```diff
@@ -439,6 +463,242 @@ public void testFollowerMappingIsUpdated() throws IOException {
+            final ClusterService clusterService = getFollowerCluster().getCurrentMasterNodeInstance(ClusterService.class);
+
+            final CountDownLatch waitForRestoreInProgress = new CountDownLatch(1);
+            final ClusterStateListener listener = event -> {
+                RestoreInProgress restoreInProgress = event.state().custom(RestoreInProgress.TYPE, RestoreInProgress.EMPTY);
+                if (restoreInProgress != null && restoreInProgress.isEmpty() == false) {
+                    waitForRestoreInProgress.countDown();
+                }
+            };
+            clusterService.addListener(listener);
+
+            final RestoreSnapshotRequest restoreRequest = new RestoreSnapshotRequest(leaderCluster, CcrRepository.LATEST)
+                .indices(leaderIndex).indicesOptions(indicesOptions).renamePattern("^(.*)$")
+                .renameReplacement(followerIndex)
+                .masterNodeTimeout(TimeValue.MAX_VALUE)
+                .indexSettings(Settings.builder()
+                    .put(IndexMetadata.SETTING_INDEX_PROVIDED_NAME, followerIndex)
+                    .put(CcrSettings.CCR_FOLLOWING_INDEX_SETTING.getKey(), true));
+            restoreService.restoreSnapshot(restoreRequest, PlainActionFuture.newFuture());
+
+            waitForRestoreInProgress.await(30L, TimeUnit.SECONDS);
+            clusterService.removeListener(listener);
+
+            final IndexRoutingTable indexRoutingTable = followerClient().admin().cluster().prepareState().clear().setRoutingTable(true)
+                .setIndices(followerIndex).get().getState().routingTable().index(followerIndex);
+            assertThat(indexRoutingTable, notNullValue());
+
+            final SnapshotsInfoService snapshotsInfoService = getFollowerCluster().getCurrentMasterNodeInstance(SnapshotsInfoService.class);
+            assertBusy(() -> {
+                SnapshotShardSizeInfo snapshotShardSizeInfo = snapshotsInfoService.snapshotShardSizes();
+                for (int shardId = 0; shardId < numberOfShards; shardId++) {
+                    final ShardRouting primary = indexRoutingTable.shard(shardId).primaryShard();
+                    assertThat(snapshotShardSizeInfo.getShardSize(primary), equalTo(ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE));
+                    final long randomSize = randomNonNegativeLong();
+                    assertThat(snapshotShardSizeInfo.getShardSize(primary, randomSize), equalTo(randomSize));
+                }
+            });
+        } finally {
+            transportServices.forEach(MockTransportService::clearAllRules);
+        }
+
+        assertThat(indicesStatsRequestsCount.get(), equalTo(numberOfShards));
+        blockCcrRestore.countDown();
+
+        followerClient().admin().cluster().prepareReroute().get();
+        ensureFollowerGreen(followerIndex);
+
+        assertAcked(followerClient().admin().indices().prepareDelete(followerIndex).setMasterNodeTimeout(TimeValue.MAX_VALUE));
```

Comment:
Why do we need `TimeValue.MAX_VALUE`, the default `30s` or so should be fine? (same in the other test?)

Reply:
Because of `CcrIntegTestCase#setupMasterNodeRequestsValidatorOnFollowerCluster()` which verifies that the master node timeout is unlimited on `AcknowledgedRequest` requests (#60070 has some background on this)